cfme/tests/infrastructure/test_webmks_console.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+"""Test for WebMKS Remote Consoles of VMware Providers."""
+import imghdr
+import pytest
+import re
+import time
+
+from cfme.common.vm import VM
+from cfme.configure import configuration
+from cfme.configure.configuration import VMwareConsoleSupport
+from cfme.infrastructure.provider import InfraProvider
+from utils import testgen, version, ssh
+from utils.appliance.implementations.ui import navigate_to
+from utils.conf import credentials
+from utils.log import logger
+from utils.providers import ProviderFilter
+from wait_for import wait_for
+
+pytestmark = pytest.mark.usefixtures('setup_provider')
+
+pytest_generate_tests = testgen.generate(
+    gen_func=testgen.providers,
+    filters=[ProviderFilter(classes=[InfraProvider], required_flags=['webmks_console'])],
+    scope='module'
+)
+
+
+@pytest.yield_fixture(scope="function")
+def vm_obj(request, provider, setup_provider, console_template, vm_name):
+    """VM creation/deletion fixture.
+
+    Create a VM on the provider with the given template, and return the vm_obj.
+    Also, remove VM from provider using nested function _delete_vm
+    after the test is completed.
+    """
+    vm_obj = VM.factory(vm_name, provider, template_name=console_template)
+    vm_obj.create_on_provider(timeout=2400, find_in_cfme=True, allow_skip="default")
+    yield vm_obj
+    try:
+        vm_obj.delete_from_provider()
+    except Exception:
+        logger.warning("Failed to delete vm `{}`.".format(vm_obj.name))
+
+
+@pytest.fixture(scope="module")
+def configure_vmware_console_for_test(appliance, provider):
+    """Configure VMware Console to use VNC which is what is required for the HTML5 console."""
+    navigate_to(appliance.server, 'Server')
+
+    settings_pg = VMwareConsoleSupport(
+        appliance=appliance,
+        console_type='VMware WebMKS',
+    )
+    settings_pg.update()
+
+
+@pytest.yield_fixture(scope="session")
+def configure_websocket():
+    """Enable websocket role if it is disabled.
+
+    Currently the fixture cfme/fixtures/base.py,
+    disables the websocket role to avoid intrusive popups. It is enabled
+    here before test begins and disabled during teardown.
+    """
+    roles = configuration.get_server_roles()
```

Comment:
**QUESTION**: Doesn't the set_roles check with the get_roles anyway? I think there is more code here than there needs to be